include/h3api/H3Managers/H3SoundManager.hpp

```diff
@@ -57,6 +57,13 @@ namespace h3
 		 * @param milliseconds The maximum duration in milliseconds, -1 defaults to 10,000ms
 		*/
 		_H3API_ VOID PlaySoundAsync(LPCSTR wav_name, INT32 duration = -1);
+        /**
+         * @brief play a sound directly.
+         * 
+         * @param wav_name The name of the file to play
+         * @return Played Wav file, WARNNING Homm3HD MOD will modify this return pointer to INT32 value as waiting timesteps
+         */
+        _H3API_ static INT64 PlaySoundByFileAsync(LPCSTR wav_name);
```

Comment:
It's not INT64 either but at least this is the correct size (:

Reply:
> It's not INT64 either but at least this is the correct size (:

Maybe a float, i guess. 😭